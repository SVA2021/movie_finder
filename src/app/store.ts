import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import authReducer from '../features/auth/authSlice';
import movieFinderReducer from '../features/movieFinder/movieFinderSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
		movieFinder: movieFinderReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
