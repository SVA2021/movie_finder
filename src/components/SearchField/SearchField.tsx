import {Search} from '@mui/icons-material';
import {Box, IconButton, Input, InputAdornment} from '@mui/material';
import {useRef} from 'react';
import {createSearchParams, useNavigate} from 'react-router-dom';
import {useAppDispatch} from '../../app/hooks';
import {setSearchKeyword} from '../../features/movieFinder/movieFinderSlice';

export const SearchField = () => {
    const ref = useRef<HTMLInputElement | undefined>();
    const navigate = useNavigate();
    const dispatch = useAppDispatch();

    function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
        e.preventDefault();
        if (!ref || !ref.current) return false;
        const keyword = ref.current.value;
        const path = {
            pathname: '/search',
            search: createSearchParams({keyword}).toString(),
        };
        navigate(path);
        dispatch(setSearchKeyword({keyword, page: 1}));
        ref.current.value = '';
    }

    return (
        <Box component={'form'} onSubmit={handleSubmit} p={1}>
            <Input
                sx={{
                    p: 2,
                    bgcolor: 'secondary.main',
                    borderRadius: 4,
                    opacity: 0.5,
                    '&:hover': {
                        opacity: 0.75,
                    },
                    '&:focus-within': {
                        opacity: 1,
                    },
                }}
                disableUnderline
                inputRef={ref}
                fullWidth
                placeholder='Movies, series, keywords...'
                endAdornment={
                    <InputAdornment position="end"  >
                        <IconButton
                            type={'submit'}
                            sx={{
                                color: 'primary.main',
                                '&:hover': {
                                    color: 'action.hover',
                                },
                                '&:active': {
                                    color: 'action.active',
                                },
                            }}
                        >
                            <Search fontSize='medium' />
                        </IconButton>
                    </InputAdornment>
                }
            />
        </Box>
    )
};