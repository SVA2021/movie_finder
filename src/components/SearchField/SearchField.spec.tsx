import {fireEvent, render, screen} from "@testing-library/react";
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {store} from "../../app/store";
import {SearchField} from "./SearchField";

describe('render SearchField', () => {

  const setUp = () => render(
    <Provider store={store}>
      <BrowserRouter>
        <SearchField />
      </BrowserRouter>
    </Provider>
  );

  it('snapshot test', () => {
    setUp();
    expect(screen.getByTestId('search-field')).toMatchSnapshot();
  });

})