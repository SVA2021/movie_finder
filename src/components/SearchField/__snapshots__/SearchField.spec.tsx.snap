// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SearchField snapshot test 1`] = `
<form
  class="MuiBox-root css-hpgf8j"
  data-testid="search-field"
>
  <div
    class="MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedEnd css-1qmitm0-MuiInputBase-root-MuiInput-root"
  >
    <input
      class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
      placeholder="Movies, series, keywords..."
      type="text"
      value=""
    />
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1wgghly-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</form>
`;
